process/src/main/java/org/openjdk/skara/process/Execution.java

```diff
@@ -138,52 +137,6 @@ private void prepareRedirects() throws IOException {
         }
     }
 
-    private void startProcessPipe() throws IOException {
-        cmd = processBuilders.stream()
-                             .map(p -> String.join(" ", p.command()))
-                             .collect(Collectors.joining(" | "));
-        log.fine("Executing pipeline '" + cmd + "'");
-
-        prepareRedirects();
-        start = Instant.now();
-
-        processes = ProcessBuilder.startPipeline(processBuilders);
-    }
-
-    private void waitForProcessPipe() throws IOException, InterruptedException {
-        var remainingTimeout = Duration.from(timeout);
-        for (var process : processes) {
-            var terminated = false;
-            try {
-                terminated = process.waitFor(timeout.toMillis(), TimeUnit.MILLISECONDS);
-                if (terminated) {
-                    var processStatus = process.exitValue();
-                    if (processStatus != 0) {
-                        // Set the final status to the rightmost command to exit with a non-zero status,
-                        // similar to pipefail in shells
-                        status = processStatus;
-                    }
-                }
-            } catch (InterruptedException e) {
-                status = -1;
-                break;
-            }
-
-            if (!terminated) {
-                log.warning("Command '" + cmd + "' didn't finish in " + timeout + ", attempting to terminate...");
-                try {
-                    process.destroyForcibly().waitFor();
-                } catch (InterruptedException e) {
-                    log.warning("Failed to terminate command");
-                    throw new RuntimeException("Failed to terminate timeouted command '" + cmd + "'");
-                }
-                throw new InterruptedException("Command '" + cmd + "' didn't finish in " + timeout + ", terminated");
-            }
-            remainingTimeout = remainingTimeout.minus(Duration.between(start, Instant.now()));
-            start = Instant.now();
-        }
-    }
-
     private void startProcess() throws IOException {
         cmd = String.join(" ", getLastProcessBuilder().command());
```

Comment:
The getLastProcessBuilder method could be removed as there is never more than one now.